checkov/secrets/git_history_store.py

```diff
@@ -0,0 +1,153 @@
+                       file_name: str, commit_hash: str, commit: Dict[str, str | Dict[str, str]]) -> None:
+        # First find if secret was moved in the file
+        equal_secret_in_commit: Dict[str, List[str]] = defaultdict(list)
+        for secret in file_results:
+            secret_key = get_secret_key(file_name, secret.secret_hash, secret.type)
+            equal_secret_in_commit[secret_key].append('added' if secret.is_added else 'removed')
+
+        for secret in file_results:
+            if secret.filename in ['None', '']:
+                secret.filename = file_name
+            secret_key = get_secret_key(file_name, secret.secret_hash, secret.type)
+            if all(value in equal_secret_in_commit[secret_key] for value in ['added', 'removed']):
+                continue
+            if secret.is_added:
+                self._add_new_secret(secret_key, commit_hash, secret, commit)
+            if secret.is_removed:
+                self._update_removed_secret(secret_key, secret, file_name, commit_hash)
+
+    def _add_new_secret(self, secret_key: str,
+                        commit_hash: str,
+                        secret: PotentialSecret,
+                        commit: Dict[str, str | Dict[str, str]]) -> None:
+        if secret_key not in self.secrets_by_file_value_type:
+            self.secrets_by_file_value_type[secret_key] = []
+        else:
+            all_removed = all(
+                potential_secret.get('removed_commit_hash') for potential_secret in
+                self.secrets_by_file_value_type[secret_key])
+            # Update secret map with the new potential secret
+            if all_removed:
+                self.secrets_by_file_value_type[secret_key][0].update(
+                    {'potential_secret': secret, 'removed_commit_hash': ''})
+                return
+        code_line = search_for_code_line(commit[secret.filename], secret.secret_value, secret.is_added)
+        self.secrets_by_file_value_type[secret_key].append(
+            {'added_commit_hash': commit_hash,
+             'removed_commit_hash': '',
+             'potential_secret': secret,
+             'code_line': code_line
+             })
+
+    def _update_removed_secret(self, secret_key: str,
+                               secret: PotentialSecret,
+                               file_name: str,
+                               commit_hash: str) -> None:
+        # Try to find the corresponding added secret in the gitz t history secret map
+        try:
+            for secret_in_file in self.secrets_by_file_value_type[secret_key]:
+                if secret_in_file['potential_secret'].is_added:
+                    secret_in_file['removed_commit_hash'] = commit_hash
+                    secret_in_file['potential_secret'] = secret
+                    break
+        except KeyError:
+            logging.error(f"No added secret commit found for secret in file {file_name}.")
```

Comment:
why not do a lookup in the map instead of catching the KeyError

Reply:
fixed